### PR TITLE
Introduce Karma+Webpack as a testing env.

### PR DESCRIPTION
**Pull request by pomek**
Fixes: #4 

It's a fundament for the other packages. This PR does not introduce any solution for manual tests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,8 +5,70 @@ CKEditor 5 Tests
...
+const ckeditor5DevTests = require( '@ckeditor/ckeditor5-dev-tests' );
+
+gulp.task( 'test', () => {
+	const path = require( 'path' );
+	
+	// "ckeditor5DevTests.utils.parseArguments" takes arguments from command line.
+	const options = ckeditor5DevTests.utils.parseArguments();
+	
+	// Base path that will be used to resolve all patterns.
+	options.rootPath = path.resolve( '.build' );
```
Comment (Reinmar) [inline README.md]: What's ".build" directory?

File: README.md [hunk 2]
```diff
@@ -5,8 +5,70 @@ CKEditor 5 Tests
...
+		// "utils.getPackageName()" returns a package name (from package.json) based on current work directory.
+		ckeditor5DevTests.utils.getPackageName()
+	];
+
+	// "ckeditor5DevTests.tests.test" returns an instance of Promise.
+	return ckeditor5DevTests.tests.test( options );
+} );
+```
+
+#### Available parameters from CLI
```
Comment (Reinmar) [inline README.md]: CLI Options

File: README.md [hunk 3]
```diff
@@ -5,8 +5,70 @@ CKEditor 5 Tests
...
+$ gulp test -c --paths=enter,paragraph
+```
+
+For testing *view* module from `ckeditor5-engine` and run the test automatically after changes in compiled code you can use:
+
+```bash
+$ gulp test -v --paths=engine/view
+```
+
+For testing specified files in `ckeditor5-basic-styles` on both browsers (Chrome and Firefox) you can use:
```
Comment (Reinmar) [inline README.md]: "on two browsers". "Both" indicates that these are the only browsers out there.

File: README.md [hunk 4]
```diff
@@ -5,8 +5,70 @@ CKEditor 5 Tests
...
+
+`gulpfile.js`
+
+```js
+const ckeditor5DevTests = require( '@ckeditor/ckeditor5-dev-tests' );
+
+gulp.task( 'test', () => {
+	const path = require( 'path' );
+	
+	// "ckeditor5DevTests.utils.parseArguments" takes arguments from command line.
```
Comment (Reinmar) [inline README.md]: Why repeating the contents of the line below in a comment?

File: README.md [hunk 5]
```diff
@@ -5,8 +5,70 @@ CKEditor 5 Tests
...
+	const options = ckeditor5DevTests.utils.parseArguments();
+	
+	// Base path that will be used to resolve all patterns.
+	options.rootPath = path.resolve( '.build' );
+
+	// Specify paths to tests. Must be relative to "tests" directory inside `options.rootPath`.
+	options.paths = [
+		// "utils.getPackageName()" returns a package name (from package.json) based on current work directory.
```
Comment (Reinmar) [inline README.md]: Again. Repeating what the code below says.

"based on the current working directory. Useful when running package tests from that package."

File: lib/utils.js [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+			// List of files to exclude.
+			exclude: [
+				// Ignore all utils which aren't tests.
+				path.join( 'tests', '**', '_utils', '**', '*.js' ),
+
+				// All manual tests.
+				path.join( 'tests', '**', 'manual', '**', '*.js' ),
+
+				// And all tickets tests (most probably they are also the manual tests).
+				path.join( 'tests', '**', 'tickets', '**', '*.js' ),
```
Comment (Reinmar) [inline lib/utils.js]: Not needed. You can remove this.

File: lib/utils.js [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+	 * @param {String} [cwd=process.cwd()] cwd Current work directory.
+	 * @throws {Error}
+	 * @returns {String}
+	 */
+	getPackageName( cwd = process.cwd() ) {
+		const packageJson = require( path.join( cwd, 'package.json' ) );
+		const matchedName = packageJson.name.match( /ckeditor5-(.*)/ );
+
+		if ( !matchedName ) {
+			throw new Error( 'Cannot find string starting with "ckeditor5-".' );
```
Comment (Reinmar) [inline lib/utils.js]: "The package name does not start with a 'ckeditor5-''

File: lib/utils.js [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+		}
+
+		return webpackConfig;
+	},
+
+	/**
+	 * Returns a name of package based on current work directory.
+	 *
+	 * @param {String} [cwd=process.cwd()] cwd Current work directory.
+	 * @throws {Error}
```
Comment (Reinmar) [inline lib/utils.js]: Really? ;)

Comment (pomek) [inline lib/utils.js]: ?

Comment (Reinmar) [inline lib/utils.js]: I mean – we don't document such things usually.

Comment (pomek) [inline lib/utils.js]: :D 
Ok, I'll remove it. My editor added this automatically.

File: README.md [hunk 9]
```diff
@@ -5,8 +5,66 @@ CKEditor 5 Tests
...
+
+`gulpfile.js`
+
+```js
+const ckeditor5DevTests = require( '@ckeditor/ckeditor5-dev-tests' );
+
+gulp.task( 'test', () => {
+	const options = ckeditor5DevTests.utils.parseArguments();
+
+	// Specify paths to tests. Must be relative to "tests" directory inside `options.rootPath`.
```
Comment (Reinmar) [inline README.md]: And how's the rootPath specified? You mention an option but you don't tell how and where it's set. It's super confusing.

Comment (Reinmar) [inline README.md]: Ooops – I've just seen it explained below :D.

File: README.md [hunk 10]
```diff
@@ -5,8 +5,66 @@ CKEditor 5 Tests
...
+
+	// Returns an instance of Promise.
+	return ckeditor5DevTests.tests.test( options );
+} );
+```
+
+#### CLI options
+
+* `watch` - Whether to watch the files and executing tests whenever any file changes. Also available as an alias: `-w`.
+* `sourceMap` - Whether to generate the source maps. Also available as an alias: `-s`.
```
Comment (Reinmar) [inline README.md]: In other words CLI options are a different thing than `test()`'s options.

Comment (pomek) [inline README.md]: Ok. Done.

**Comment by Reinmar** [comment]
Sounds like we did it! :) All works fine for me.

**Pull request merged**